CONTRIBUTING.md

```diff
@@ -0,0 +1,147 @@
+each make small but meaningful changes. Striking a balance between minimal
+commits and logically complete changes is an art as much as a science, but
+when it is possible and reasonable, divide your pull request into more commits.
+
+Some times when it will almost always make sense to separate parts of a change
+into their own commits are:
+- Changes to unrelated formatting and typo-fixing.
+- Refactoring changes that prepare the codebase for your logical change.
+- Similar changes to multiple parts of the codebase (for instance, similar
+  changes to handling of CephFilesystem, CephBlockPool, and CephObjectStore).
+
+Even when breaking down commits, each commit should leave the codebase in a
+working state. The code should add necessary unit tests and pass unit tests,
+formatting tests, and usually functional tests. There can be times when
+exceptions to these requirements are appropriate (for instance, it is sometimes
+useful for maintainability to split code changes and related changes to CRDs
+and CSVs). Unless you are very sure this is true for your change, though, make
+sure each commit passes CI checks as above.
+
+### Commit and Pull Request Messages
+
+*This requirement cannot be fully tested by make ocs-operator-ci.*
+
+- The message must begin with a single summary line describing the change.
+  - It must be capitalized.
+  - It must not end with a period.
+  - It must be <= 80 characters in length.
+  - *Recommendation*: It should be written in the imperative\ mood: "Fix bug x"
+    rather than "Fixed bug x" or "Fixes bug x". (The sentence "If applied, this
+    commit will \<your summary\>" should be grammatical.)
+- The message should continue with a longer description of the change.
+  - The description must be preceded by a single blank line to separate it from
+    the summary.
+  - It must describe why the change is necessary or useful.
+  - It must describe how the change was implemented.
+  - It must reference any open Issues the change addresses.
+  - It must wrap at 80 characters.
+  - *For very small changes it may be acceptable to omit the longer description.
+    Please remember, that it is easy for any developer to believe their code is
+    "self-documenting" when it is not. Adding a description will help future
+    maintainers of the project.*
+- The message must end with a sign-off, as discussed in [Certificate of
+  Origin](#certificate-of-origin).
+  - The sign-off must be preceded by a single blank line to separate it from
+    the preceding section. 
+  - If the code has multiple authors:
+     - Each author must add a sign-off.
+     - *Recommendation*: A "Co-authored-by:" line should be added for each
+       additional author.
+
+Pull request messages should follow the same format, but for the entire set of
+changes contained in the pull request.
+
+### Coding Style
+
+All changes must follow style guidelines tested by the `golangci/golangci-lint
```

Comment:
```suggestion
All changes must follow style guidelines tested by the `golangci/golangci-lint`
```

Reply:
Foolishness on my part. Thanks! And yes, we definitely had some good discussions up there.

Reply:
:-)